src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1930,21 +1923,22 @@ oop java_lang_Thread::async_get_stack_trace(oop java_thread, TRAPS) {
         return;
       }
 
-      JavaThread* thread = JavaThread::cast(th);
+      JavaThread* java_thread = JavaThread::cast(th);
 
-      if (!thread->has_last_Java_frame()) {
+      if (!java_thread->has_last_Java_frame()) {
         return;
       }
 
       bool carrier = false;
-      if (java_lang_VirtualThread::is_instance(_java_thread())) {
-        // if (thread->vthread() != _java_thread()) // We might be inside a System.executeOnCarrierThread
-        const ContinuationEntry* ce = thread->vthread_continuation();
-        if (ce == nullptr || ce->cont_oop(thread) != java_lang_VirtualThread::continuation(_java_thread())) {
-          return; // not mounted
+      if (java_lang_VirtualThread::is_instance(_thread_oop())) {
+        // Ensure _thread_oop is still mounted to java_thread.
+        const ContinuationEntry* ce = java_thread->vthread_continuation();
+        if (ce == nullptr || ce->cont_oop(java_thread) != java_lang_VirtualThread::continuation(_thread_oop())) {
+          // Target thread has been unmounted.
+          return;
         }
```

Comment:
Shouldn't we set `carrier = true` if we don't return here? Or am I misunderstanding what "carrier" means here?

Reply:
"carrier" means target is a platform thread with mounted virtual thread.
When `carrier == true` `vframeStream` skips frames of the mounted virtual thread.
Here target is a virtual thread, so we need frames starting from the last frame